classes/Category.php

```diff
@@ -872,8 +872,8 @@ public static function getSimpleCategoriesWithParentInfos($idLang)
 		ON (c.`id_category` = cl.`id_category`' . Shop::addSqlRestrictionOnLang('cl') . ')
 		' . Shop::addSqlAssociation('category', 'c') . '
 		WHERE cl.`id_lang` = ' . (int) $idLang . '
-        AND c.`nleft` >= ' . (int) $rootTreeInfo['nleft'] . '
-        AND c.`nright` <= ' . (int) $rootTreeInfo['nright'] . '
+        AND c.`nleft` >= ' . (int) ($rootTreeInfo['nleft'] ?? 0) . '
```

Comment:
This change is a bit weird no? 🤔

It means the filter as no impact any more, that the tree has no root which should never happen If there is no root then the function should return before getting here. Maybe PHP version had different behaviours and casted this automatically as null, then zero but it's not a good reason to ignore an invalid behaviour by just adding some `??`

Reply:
Nope... Why ?

It was broken in PHP 7.4... but hidden in tests-legacy.

In tests in PHP 7.3/7.2 : you have `$rootTreeInfo` which was `null` so the value casted was 0. So I apply the same behavior in PHP 7.4.

So if the behavior change it should be in a new PR.

Reply:
That's your modif which is changing the behaviour I understand that the previous assumptions of the language was hiding the bug, no problem with that But then replicating the error of the language is not a bug fix

It's just hiding the problem under the carpet, the real bug is not the forced cast It's the absent root tree! So either the test environment is not correct and thus returns empty root tree, or the absence of the tree indicates another issue I don't know but I guess it requires further research

Reply:
Actually, I keep the initial behavior and I don't hide me on this point. If we need to adapt tests and code, we need to do it in a specific PR for only targetting the problem.

Reply:
Better check if `$rootTreeInfo` is not empty, if yes ,return before running this SQL query

Reply:
But if there is no rootTree then the search in the nested tree has no meaning.. So instead we should check above something like:

```php
if (empty($rootTreeInfo)) {
    return [];
}
```

(or `false` or `null` I don't know the expected result when no tree is found)

Reply:
Fixed